plugins/l4lb/l4lb.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2015 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"runtime"
+
+	"github.com/dcos/dcos-cni/pkg/spartan"
+
+	"github.com/containernetworking/cni/pkg/invoke"
+	"github.com/containernetworking/cni/pkg/ip"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/version"
+)
+
+type NetConf struct {
+	types.NetConf
+	Args     map[string]interface{} `json:"args"`
+	MTU      int                    `json:"mtu"`
+	Delegate map[string]interface{} `json:"delegate"`
+}
+
+func init() {
+	// This ensures that main runs only on main thread (thread group leader).
+	// Since namespace ops (unshare, setns) are done for a single thread, we
+	// must ensure that the goroutine does not jump from OS thread to thread
+	runtime.LockOSThread()
+}
+
+func setupDelegateConf(conf *NetConf) (delegateConf []byte, delegatePlugin string, err error) {
```

Comment:
mutable parameters are allowed. in this case, it's more idiomatic for `conf` to be a receiver -- it's the only mutable thing here